### PR TITLE
[Fix] Remove markedForInsertion, markedForDeletion  from Role

### PR DESCRIPTION
**Pull request by KaterinaWire**
### Issues
We don't use markedForInsertion and markedForDeletion for Roles.

### Causes
To remove or insert roles, we send the request directly to BE.

### CODE REVIEW

**Comment by zenkins** [comment]
Can one of the admins verify this patch?

File: Tests/Source/Model/Observer/ConversationObserverTests.swift [hunk 1]
```diff
@@ -347,57 +347,57 @@ final class ConversationObserverTests : NotificationDispatcherTestBase {
...
-                                                    }
-                                                },
-                                                expectedChangedFields: ["participantsChanged",
-                                                     "nameChanged",
-                                                     "activeParticipantsChanged"],
-                                                expectedChangedKeys: ["displayName",
-                                                                      "localParticipantRoles"])
-        
-    }
+//    func testThatItNotifiesTheObserverOfAParticipantOperationSetToDelete() {
```
Comment (johnxnguyen) [inline Tests/Source/Model/Observer/ConversationObserverTests.swift]: Are you going to delete these?

Comment (KaterinaWire) [reply Tests/Source/Model/Observer/ConversationObserverTests.swift]: Yes.
I deleted it.

File: Tests/Source/Model/Observer/ConversationObserverTests.swift [hunk 2]
```diff
@@ -431,7 +379,7 @@ final class ConversationObserverTests : NotificationDispatcherTestBase {
         let conversation = ZMConversation.insertNewObject(in: self.uiMOC)
         conversation.conversationType = ZMConversationType.group
         conversation.addParticipantAndUpdateConversationState(user: user, role: nil)
-        conversation.participantRoles.forEach { $0.operationToSync = .none }
+//        conversation.participantRoles.forEach { $0.operationToSync = .none }
```
Comment (johnxnguyen) [inline Tests/Source/Model/Observer/ConversationObserverTests.swift]: Don't forget to delete this.

File: Tests/Source/Model/Observer/ConversationObserverTests.swift [hunk 3]
```diff
@@ -466,7 +414,7 @@ final class ConversationObserverTests : NotificationDispatcherTestBase {
         let conversation = ZMConversation.insertNewObject(in: self.uiMOC)
         conversation.conversationType = ZMConversationType.group
         conversation.addParticipantAndUpdateConversationState(user: user, role: oldRole)
-        conversation.participantRoles.forEach { $0.operationToSync = .none }
+//        conversation.participantRoles.forEach { $0.operationToSync = .none }
```
Comment (johnxnguyen) [inline Tests/Source/Model/Observer/ConversationObserverTests.swift]: Don't forget to delete this.

File: Tests/Source/Model/Observer/ConversationObserverTests.swift (hunk 2 above)
Comment (KaterinaWire) [reply Tests/Source/Model/Observer/ConversationObserverTests.swift]: Fixed

File: Tests/Source/Model/Observer/ConversationObserverTests.swift (hunk 3 above)
Comment (KaterinaWire) [reply Tests/Source/Model/Observer/ConversationObserverTests.swift]: Fixed

**Pull request merged**